File: cs/Problems/Day4Test.cs
namespace aoc24.Problems;

public sealed class Day4Test
{
    private readonly Day4 sut = new();

    [Theory, InlineData(18)]
    public void TestSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day4_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(2575)]
    public void FullSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day4_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day4_test.go
package problems

import "testing"

func TestDay4_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day4{}, "day4_1", 18)
}

func TestDay4_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day4{}, "day4_2", 2575)
}
